### DISCUSSION TITLE
yt-dlp wrapper class for youtube download not working in cpanel php script

### DISCUSSION THREAD

**Discussion by SupMaulik** [norkunas/youtube-dl-php#206] [General]
I am using yt-dlp in my domain racer c panel php script to download youtube video . but yt-dlp not working. 
i have written folowing code in cpanel  public_html
<?php

declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

use YoutubeDl\Options;
use YoutubeDl\YoutubeDl;

$yt = new YoutubeDl();
$yt->setBinPath('yt-dlp');
$collection = $yt->download(
    Options::create()
        ->downloadPath('/')
        ->url('https://www.youtube.com/watch?v=oDAw7vW7H0c')
);

foreach ($collection->getVideos() as $video) {
    if ($video->getError() !== null) {
        echo "Error downloading video: {$video->getError()}.";
    } else {
        print_r($video); // Will return Phonebloks
        // $video->getFile(); // \SplFileInfo instance of downloaded file
    }
}
why yt-dlp not working on cpanel ?

**Comment by norkunas** [comment]
Without debugging I cannot say why it's not working.
Use:
```
$yt->debug(function ($type, $buffer) {
    if (\Symfony\Component\Process\Process::ERR === $type) {
        echo "err: $buffer";
    } else {
        echo $buffer;
    }
});
```
to see what's going on

**Comment by SupMaulik** [comment]
Thanks for reply. I have got solution. i have corrected setbinpath . now
everything working fine .
                                                             Thank and
regards ,Suprabhat Joshi

**Comment by SupMaulik** [comment]
Hello Sir,
                       I need help
<?php

// Command to get the video info in JSON format using Python



$command = 'yt-dlp -f "bestvideo+bestaudio" --get-url "
https://www.youtube.com/watch?v=Zt5O1JN59Vo"';
$link = shell_exec($command);

echo "<pre>";
print_r($link);
echo "</pre>";


?>
i am using this php script to get one merge url link for 720p with
video and audio
but it give two separate url link one for video and one for audio like below

https://rr4---sn-cvh7knzl.googlevideo.com/videoplayback?expire=1719182778&ei=WlF4Ztz7Ooe7rtoP48Cs0AE&ip=2401%3A4900%3A1f3b%3A7b65%3Ac423%3A5b8f%3A59cf%3A4ad0&id=o-ABrQQLNPhaS5rXAeId2vb0sib4Ts0dFOt4bE82j4XLn9&itag=137&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=C-&mm=31%2C29&mn=sn-cvh7knzl%2Csn-ci5gup-qxar&ms=au%2Crdu&mv=m&mvi=4&pl=54&initcwndbps=1602500&vprv=1&svpuc=1&mime=video%2Fmp4&rqh=1&gir=yes&clen=301084528&dur=914.213&lmt=1689445382909251&mt=1719160868&fvip=5&keepalive=yes&c=IOS&txp=6219224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAOSlmWpJsoar5sPgYBoeACb_fAoCVxiSBoUVla1WNEAxAiB57hd7Ser9AsIZCEDocHqyBnjWeK_DhEQ5GAmEMNi49A%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHlkHjAwRgIhAOEsblrzHVtKAbFB4nZWguuJNrE1ZwCYCDEba3hYu4etAiEA_8k9ohYDwLmyqP8q6epz-YaVRfqyF-y1TgWQbIEDW2c%3Dhttps://rr5---sn-cvhelnls.googlevideo.com/videoplayback?expire=1719182778&ei=WlF4ZsedD5WpjuMPx66i8Ag&ip=2401%3A4900%3A1f3b%3A7b65%3Ac423%3A5b8f%3A59cf%3A4ad0&id=o-AFryiBxoR6xYMacbjAzpH0xENXyJ6TzpJjBQOVr0MTUa&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=C-&mm=31%2C29&mn=sn-cvhelnls%2Csn-ci5gup-qxar&ms=au%2Crdu&mv=m&mvi=5&pl=54&initcwndbps=1592500&bui=AbKP-1MxC4U63jS873_om3_C3gb3f_je-xttp7HvgJYJrqJQFVReLjWJJAjX74nJMEao7irWtvGLViOp&spc=UWF9f-USKYjOrt5EXPp2UsFbRysGxxpuMAnDJ3ji5MiGMp-vOQFsYraeaxvX&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=QRCK6OdGlZULkKd343fcg-sQ&rqh=1&gir=yes&clen=12888341&dur=914.241&lmt=1689445389573221&mt=1719160868&fvip=5&keepalive=yes&c=WEB&sefc=1&txp=6218224&n=59CRjPyMwfzytQ&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRAIgEeujxJt3KOKCPpY_avXuQ3perMxai4SzerOHBDZs_yACIDCZbTKKhae0Fv7AC5vjW_OwI7zIpTAoH2g3Gh8ViF13&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHlkHjAwRgIhAI0TYwAQXYhjeXsOV6dYc9ACP1hEtfWMtAWIDm94yevAAiEArvyZ8Ai03SbflkVHz5ICKj1H-8XGlXfWmOBaX6WiF6w%3D

how can i get merge url single link to play 720p video with audio

**Reply by norkunas** [reply]
Hi,

I'm not sure how this is related to this library 🤔
If things are specific you the ytdl itself, I suggest to ask on their repositories :)
